src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs

```diff
@@ -102,7 +103,9 @@ private void updateOutputMappingList()
 
             var newOutputMappingDTO = mapFrom(newOutputMapping);
 
-            _simulation.OutputMappings.Add(newOutputMapping);
+            if (newOutputMapping.Output != null) 
+               _simulation.OutputMappings.Add(newOutputMapping);
```

Comment:
Let's review this code here and the one below. I don not understand what's going on. You are creating a cache of all output in a loop where the cache is independent form the variable being changed. We are almost there but this code is not efficient nad we should check this out one more time

Reply:
@msevestre we can talk about this today in the meeting (or even later if you prefer)